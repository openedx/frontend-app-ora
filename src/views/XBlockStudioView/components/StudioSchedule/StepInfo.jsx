import React from 'react';
import PropTypes from 'prop-types';

import { useIntl } from '@edx/frontend-platform/i18n';

import FormatDateTime from './FormatDateTime';

import messages from '../messages';

const StepInfo = ({ stepName, startDatetime, endDatetime }) => {
  const { formatMessage } = useIntl();
  return (
    <div>
      {startDatetime && (
        <p>
          <strong>{stepName} {formatMessage(messages.startLabel)}</strong>
          <FormatDateTime date={startDatetime} />
        </p>
      )}
      {endDatetime && (
        <p>
          <strong>{stepName} {formatMessage(messages.dueLabel)}</strong>
          <FormatDateTime date={endDatetime} />
        </p>
      )}
    </div>
  );
};

StepInfo.defaultProps = {
  startDatetime: null,
  endDatetime: null,
};

StepInfo.propTypes = {
  stepName: PropTypes.string.isRequired,
  startDatetime: PropTypes.string,
  endDatetime: PropTypes.string,
};

export default StepInfo;
