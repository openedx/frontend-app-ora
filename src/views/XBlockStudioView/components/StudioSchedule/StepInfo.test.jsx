import { shallow } from '@edx/react-unit-test-utils';

import StepInfo from './StepInfo';

jest.mock('./FormatDateTime', () => 'FormatDateTime');

describe('<StepInfo />', () => {
  it('should render', () => {
    const wrapper = shallow(<StepInfo stepName="test" />);
    expect(wrapper.snapshot).toMatchSnapshot();
  });

  it('should render with startDatetime and endDatetime', () => {
    const wrapper = shallow(<StepInfo stepName="test" startDatetime="2020-01-01T00:00:00Z" endDatetime="2020-01-01T00:00:00Z" />);
    expect(wrapper.snapshot).toMatchSnapshot();
  });
});
