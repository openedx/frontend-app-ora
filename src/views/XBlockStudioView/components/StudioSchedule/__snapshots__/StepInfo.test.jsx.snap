// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StepInfo /> should render 1`] = `<div />`;

exports[`<StepInfo /> should render with startDatetime and endDatetime 1`] = `
<div>
  <p>
    <strong>
      test
       
      start: 
    </strong>
    <FormatDateTime
      date="2020-01-01T00:00:00Z"
    />
  </p>
  <p>
    <strong>
      test
       
      due: 
    </strong>
    <FormatDateTime
      date="2020-01-01T00:00:00Z"
    />
  </p>
</div>
`;
